### PR TITLE
fix: mobile tooltip style

### LINKED ISSUE #6512: Mobile tooltip update gallery
Please @kodadot/internal-dev do refactor as well as needed

chains are perfect now! thanks to everyone involved

but oops, seems like mobile is wrong again:

image

_Originally posted by @prury in https://github.com/kodadot/nft-gallery/issues/6448#issuecomment-1660269992_

### PR DESCRIPTION
**Pull request by floyd-li**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

the reason that the `tooltip` is displayed in incorrect position is that the `buy button` is not centerd, so we must calc the position to decrease the `cart button`'s width so that `tooltip` can be centered

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #6512
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=13rv1SWoLg9Gb3tmvHPZxb7JbVy51BtMziX7k9WQGSJ7Kp3A)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

图片

### <samp>🤖 Generated by Copilot at b7df21b</samp>

Refactor the tip component for gallery items to improve responsiveness and alignment. Add a new modifier class `o-tip__content--top` for mobile layout.

### <samp>🤖 Generated by Copilot at b7df21b</samp>

> _`o-tip` refactored_
> _Tooltip above gallery_
> _Autumn leaves falling_

### CODE REVIEW

**Review by roiLeo — COMMENTED** [review]
wouldn't be better to move this code on `NeoTooltip` so it won't reproduce?

**Comment by floyd-li** [comment]
> wouldn't be better to move this code on `NeoTooltip` so it won't reproduce?

i'm afraid we can't. every usage of `NeoTooltip` is different and the incorrect display issue is caused by the position of the tooltip trigger component. so i think it it may not possible to create a common logic to solve it.

**Comment by prury** [comment]
Once tooltip gets activated, it never goes away

**Comment by floyd-li** [comment]
> Once tooltip gets activated, it never goes away

just checked it on canary.kodadot.xyz and it's same behavior.

seems it related with this logic

`:auto-close="!isMobileDevice ? ['outside', 'escape'] : []"`

https://github.com/kodadot/nft-gallery/blob/872e7fdcc8cdcac1c2558a0c0c4f0279e59f329a/components/gallery/GalleryItemAction/GalleryItemActionType/GalleryItemBuy.vue#L6-L12

@daiagi is it for special purpose? or should i change it as @prury expected?

**Comment by daiagi** [comment]
Feel free to change it to suit the requirements

**Comment by prury** [comment]
> Feel free to change it to suit the requirements
> 
> Keep in mind that tooltip usually don't have much value in mobile as there is no hover 

In this specific case i can trigger the tooltip on mobile by pressing and holding the deactivated buy button, but yeah, prob nobody uses it

**Review by roiLeo — APPROVED** [review]

**Comment by floyd-li** [comment]
update: fixed the offer disabled tooltip

cc @prury 

图片

File: components/gallery/GalleryItemAction/GalleryItemActionType/GalleryItemBuy.vue [hunk 1]
```diff
@@ -8,7 +8,8 @@
             class="w-full"
             content-class="buy-tooltip"
             :position="isMobileDevice ? 'top' : 'left'"
-            :auto-close="!isMobileDevice ? ['outside', 'escape'] : []"
+            :auto-close="['outside', 'escape']"
+            :triggers="[isMobileDevice ? 'click' : 'hover']"
```
Comment (daiagi) [inline components/gallery/GalleryItemAction/GalleryItemActionType/GalleryItemBuy.vue:12]: this seems like generally good logic
can it be placed as default behavior in `NeoTooltip`?

Comment (floyd-li) [reply components/gallery/GalleryItemAction/GalleryItemActionType/GalleryItemBuy.vue:12]: @daiagi  yes did it. a little question that currently we use this logic to judge if it's mobile device
```
const isMobileDevice = computed(() => useWindowSize().width.value < 1024)
```
can i import `useWindowSize` from `@vueuse/core` directly in `NeoTooltip.vue`?

File: components/gallery/GalleryItemTabsPanel/GalleryItemTabsPanel.vue [hunk 2]
```diff
@@ -17,6 +17,7 @@
         <NeoTooltip
           v-if="offersDisabled"
           :label="$t('tabs.offersDisabled')"
+          content-class="offers-disabled-tooltip"
```
[the remark this answers is not in the document]
Comment (floyd-li) [reply components/gallery/GalleryItemTabsPanel/GalleryItemTabsPanel.vue:20]: hmmm tried that, but seems it looks not that good
图片

**Review by daiagi — APPROVED** [review]

File: components/gallery/GalleryItemAction/GalleryItemActionType/GalleryItemBuy.vue (hunk 1 above)
Comment (floyd-li) [reply components/gallery/GalleryItemAction/GalleryItemActionType/GalleryItemBuy.vue:11]: done

**Comment by exezbcz** [comment]
@yangwao can I merge? 🥺

**Comment by yangwao** [comment]
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 4.46 USD/DOT ~ 6.726 $DOT
🧗 [13rv1SWoLg9Gb3tmvHPZxb7JbVy51BtMziX7k9WQGSJ7Kp3A](https://polkadot.subscan.io/account/13rv1SWoLg9Gb3tmvHPZxb7JbVy51BtMziX7k9WQGSJ7Kp3A)
🔗 [0x3bbb308b1009af3d87b5b1d1c033b74ab73632a70553893d4e4bc73446af8b4e](https://polkadot.subscan.io/extrinsic/0x3bbb308b1009af3d87b5b1d1c033b74ab73632a70553893d4e4bc73446af8b4e)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**